interfaces/testbackend.go

```diff
@@ -0,0 +1,67 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package interfaces
+
+import (
+	"github.com/ubuntu-core/snappy/snap"
+)
+
+// TestSecurityBackend is a security backend intended for testing.
+type TestSecurityBackend struct {
+	// SetupCalls stores information about all calls to Setup
+	SetupCalls []SetupCall
+	// RemoveCalls stores information about all calls to Remove
+	RemoveCalls []string
+	// SetupCallback is an callback that is optionally called in Setup
+	SetupCallback func(snapInfo *snap.Info, developerMode bool, repo *Repository) error
+	// RemoveCallback is a callback that is optionally called in Remove
+	RemoveCallback func(snapName string) error
+}
+
+// SetupCall stores details about calls to TestSecurityBackend.Setup
+type SetupCall struct {
+	// SnapInfo is a copy of the snapInfo argument to a particular call to Setup
+	SnapInfo *snap.Info
+	// DeveloperMode is a copy of the developerMode argument to a particular call to Setup
+	DeveloperMode bool
```

Comment:
We've used `DevMode` in the state, and the term seems like a nice short hand to the concept. We should probably even change `dev-mode` there to `devmode`, and have the command line flag as `--devmode` too.